_posts/2025-03-26-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,700 @@
+- You need to annotate _everything_ you don't want to automatically update.
+- Until a specific Kafka resource is finally upgraded, that resource is _unmanaged_. Strimzi will not reconcile it, fix any issues or perform any admin operations.
+- Roll back is an _all or nothing_ operation. If you encounter an issue you would need to pause _all_ Kafka resources again, downgrade the Strimzi Operator and then un-pause the resources.   
+
+#### Resource Selector
+
+While the options above work in theory, in practice they have several drawbacks which make them unsuitable for production deployments. 
+A better option comes by using the _Custom Resource Selector_. 
+This is an [environment variable](https://strimzi.io/docs/operators/latest/deploying#ref-operator-cluster-str) that is set in the Cluster Operator deployment:
+
+>`STRIMZI_CUSTOM_RESOURCE_SELECTOR` 
+   Optional. The label selector to filter the custom resources handled by the Cluster Operator. The operator will operate only on those custom resources that have the specified labels set. Resources without these labels will not be seen by the operator. The label selector applies to `Kafka`, `KafkaConnect`, `KafkaBridge`, `KafkaMirrorMaker`, and `KafkaMirrorMaker2` resources. `KafkaRebalance` and `KafkaConnector` resources are operated only when their corresponding Kafka and Kafka Connect clusters have the matching labels.  
+> ```yaml 
+> env:   
+> 	- name: STRIMZI_CUSTOM_RESOURCE_SELECTOR     
+> 	  value: label1=value1,label2=value2 
+> ```
+
+Using this selector you can set a label, `strimzi-0-43-0` for example, in an all-namespaces scoped Strimzi Deployment and then tag all the Kafka CRs (and other operands) with this label and that Strimzi deployment will manage them and only them. 
+Any operand CRs without the label will be ignored. 
+
+You could then deploy _another_ cluster-scoped Strimzi operator with a different custom resource selector, `strimzi-0-43-0-b` for example, and tag a different set of Operand CRs with that label.
+The two Strimzi deployments would coexist happily (provided you have setup independent RBAC -- more on this later), managing completely separate sets of CRs.
+
+This features allows us to create a more robust upgrade process for the Strimzi Operator and its operands.
+
+### Resource Selector based upgrade
+
+Using the Custom Resource Selector allows us to have fine grained control over which operands are upgraded to new Strimzi versions and when. 
+However, in order to use this process we need to deploy multiple versions of Strimzi concurrently.
+That requires us to consider several key aspects of the Strimzi installation resources:
+
+#### Concurrent Strimzi Deployments
+
+##### Custom Resource Definitions
+
+The first major consideration when designing a deployment strategy for multiple Strimzi versions is the [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) (CRDs). 
```

Comment:
You already introduced the CRDs in a previous paragraph, so you can link the doc there, and there is no need to define the acronym again.